pkg/glob/glob.go

```diff
@@ -142,7 +150,12 @@ func glob(segs []Segment, dir string, cb func(string) bool) bool {
 	for _, info := range infos {
 		name := info.Name()
 		if matchElement(segs, name) {
-			if !cb(dir + name) {
+			name = dir + name
```

Comment:
Do not modify `name` in place; it gets confusing.

Reply:
Agreed. Changed to `dirname`.